### PR TITLE
Added missing web3 functionalities from the standard

### PR DESCRIPTION
**Pull request by BatiGencho**

### CODE REVIEW

File: test/simple-browser-extension/plugin/lib/massa-wallet-provider-content-script.js [hunk 1]
```diff
@@ -124,22 +130,65 @@
...
+              const rollOperationPayload = payload.params;
+              const respMessage = {
+                  result: await this.sendTransaction(rollOperationPayload),
+                  error: null,
+                  requestId: payload.requestId,
+              };
+              // answer to the message target
+              walletProviderEventTarget.dispatchEvent(new CustomEvent('message', detailWrapper({ detail: respMessage })));
+          });
+          // ==============================GET NODE URLS==================================
```
Comment (Elli610) [inline test/simple-browser-extension/plugin/lib/massa-wallet-provider-content-script.js:178]:
```suggestion
          // ==============================GET NODES URLS==================================
```

**Review by Elli610 — APPROVED** [review]
Looks good

File: test/simple-browser-extension/plugin/lib/massa-wallet-provider-content-script.js (hunk 1 above)
Comment (BatiGencho) [reply test/simple-browser-extension/plugin/lib/massa-wallet-provider-content-script.js]: thanks, fixed.

File: src/account/Account.ts [hunk 2]
```diff
@@ -98,4 +100,77 @@ export class Account implements IAccount {
...
+  /**
+   * This method aims to buy rolls on behalf of the sender.
+   *
+   * @param amount - The amount of rolls to be purchased
+   * @param fee - The fee to be paid for the transaction.
+   * @returns An ITransactionDetails object. It contains the operationId on the network.
+   */
+  public async buyRolls(
+    amount: string,
+    fee: string,
```
Comment (gregLibert) [inline src/account/Account.ts:113]: Why not using BigInt as desribed in https://github.com/massalabs/massa-standards/blob/main/units.md ?

File: src/account/Account.ts [hunk 3]
```diff
@@ -98,4 +100,77 @@ export class Account implements IAccount {
...
+
+  /**
+   * This method aims to buy rolls on behalf of the sender.
+   *
+   * @param amount - The amount of rolls to be purchased
+   * @param fee - The fee to be paid for the transaction.
+   * @returns An ITransactionDetails object. It contains the operationId on the network.
+   */
+  public async buyRolls(
+    amount: string,
```
Comment (gregLibert) [inline src/account/Account.ts:112]: Should be an unsigned integer if that exists

File: src/account/Account.ts [hunk 4]
```diff
@@ -98,4 +100,77 @@ export class Account implements IAccount {
...
+  /**
+   * This method aims to sell rolls on behalf of the sender.
+   *
+   * @param amount - The amount of rolls to be sold.
+   * @param fee - The fee to be paid for the transaction.
+   * @returns An ITransactionDetails object. It contains the operationId on the network.
+   */
+  public async sellRolls(
+    amount: string,
+    fee: string,
```
Comment (gregLibert) [inline src/account/Account.ts:137]: Same

File: src/account/Account.ts [hunk 5]
```diff
@@ -98,4 +100,77 @@ export class Account implements IAccount {
       );
     });
   }
+
+  /**
+   * This method aims to buy rolls on behalf of the sender.
+   *
+   * @param amount - The amount of rolls to be purchased
+   * @param fee - The fee to be paid for the transaction.
```
Comment (gregLibert) [inline src/account/Account.ts:108]: We are targeting dev not used to build on blockchain. we need to be clear on for who the fees are :-)
```suggestion
   * @param fee - The fee to be paid for the transaction execution by the node.
```

File: src/account/Account.ts [hunk 6]
```diff
@@ -98,4 +100,77 @@ export class Account implements IAccount {
...
+        },
+      );
+    });
+  }
+
+  /**
+   * This method aims to sell rolls on behalf of the sender.
+   *
+   * @param amount - The amount of rolls to be sold.
+   * @param fee - The fee to be paid for the transaction.
```
Comment (gregLibert) [inline src/account/Account.ts:132]: same

File: src/account/Account.ts [hunk 7]
```diff
@@ -98,4 +100,77 @@ export class Account implements IAccount {
...
+
+  /**
+   * This method aims to transfer MAS on behalf of the sender to a recipient.
+   *
+   * @param amount - The amount of MAS to be transferred.
+   * @param fee - The fee to be paid for the transaction.
+   * @returns An ITransactionDetails object. It contains the operationId on the network.
+   */
+  public async sendTransaction(
+    amount: string,
```
Comment (gregLibert) [inline src/account/Account.ts:160]: Same as fee.

File: src/provider/Provider.ts [hunk 8]
```diff
@@ -144,4 +145,25 @@ export class Provider implements IProvider {
...
     });
   }
+
+  /**
+   * This method generates a new account by a given name and adds it to the wallet.
+   *
+   * @param name - The account name
+   * @returns a Promise that resolves to an instance of IAccountDetails.
+   */
+  public async generateNewAccount(name: string): Promise<IAccountDetails> {
```
Comment (gregLibert) [inline src/provider/Provider.ts:155]:
```suggestion
  public async generateNewAccount(name?: string): Promise<IAccountDetails> {
```

File: src/thyra/ThyraAccount.ts [hunk 9]
```diff
@@ -140,4 +142,41 @@ export class ThyraAccount implements IAccount {
...
+
+  /**
+   * This method aims to buy rolls on behalf of the sender.
+   *
+   * @param amount - The amount of rolls to be bought.
+   * @param fee - The fee to be paid for the transaction.
+   * @returns An ITransactionDetails object. It contains the operationId on the network.
+   */
+  buyRolls(amount: string, fee: string): Promise<ITransactionDetails> {
+    throw new Error('Method not implemented.');
```
Comment (gregLibert) [inline src/thyra/ThyraAccount.ts:154]: Can you create an issue on Thyra side if this is not yet done ?

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: src/account/Account.ts (hunk 2 above)
Comment (BatiGencho) [reply src/account/Account.ts]: fixed

File: src/account/Account.ts (hunk 3 above)
Comment (BatiGencho) [reply src/account/Account.ts]: we dont have that in typescript, hence a bigint would suffice here. I could also add a check > 0 if you wish ?

File: src/account/Account.ts (hunk 4 above)
Comment (BatiGencho) [reply src/account/Account.ts]: done

File: src/account/Account.ts (hunk 5 above)
Comment (BatiGencho) [reply src/account/Account.ts]: changed

File: src/account/Account.ts (hunk 6 above)
Comment (BatiGencho) [reply src/account/Account.ts]: changed

File: src/account/Account.ts (hunk 7 above)
Comment (BatiGencho) [reply src/account/Account.ts]: done

File: src/provider/Provider.ts (hunk 8 above)
Comment (BatiGencho) [reply src/provider/Provider.ts]: changed

File: src/thyra/ThyraAccount.ts (hunk 9 above)
Comment (BatiGencho) [reply src/thyra/ThyraAccount.ts:154]: Sure, I will ,but need to ask the inno team about this

**Review by gregLibert — APPROVED** [review]

**Pull request merged**